File: docs/PayTR Linkle Ödeme/Türkçe/PayTR Link API - Callback (Optional)/Türkçe/Örnek PHP/PayTR Link API - Callback.php
<?php

    ############################# ÖDEME LİNKİ BİLDİRİM ÖRNEK KODLAR ############################
    #                                                                                          #
    $post = $_POST;

    ################################ DÜZENLEMESİ ZORUNLU ALANLAR ###############################
    #
    ## API Entegrasyon Bilgileri - Mağaza paneline giriş yaparak BİLGİ sayfasından alabilirsiniz.
    $merchant_key   = 'XXXXXXXXXXXXXXXX';
    $merchant_salt  = 'XXXXXXXXXXXXXXXX';
    ############################################################################################

    ################ Bu kısımda herhangi bir değişiklik yapmanıza gerek yoktur. ################
    #
    ## POST değerleri ile hash oluştur.
    $hash = base64_encode( hash_hmac('sha256', $post['callback_id'].$post['merchant_oid'].$merchant_salt.$post['status'].$post['total_amount'], $merchant_key, true) );
    #
    ## Oluşturulan hash'i, paytr'dan gelen post içindeki hash ile karşılaştır (isteğin paytr'dan geldiğine ve değişmediğine emin olmak için)
    ## Bu işlemi yapmazsanız maddi zarara uğramanız olasıdır.
    if( $hash != $post['hash'] )
        die('PAYTR notification failed: bad hash');
    ############################################################################################
    #
    ## BURADA YAPILMASI GEREKENLER
    ## 1) Ödeme durumunu $post['callback_id'] değerini kullanarak veri tabanınızdan sorgulayın.
    ## 2) Eğer ödeme zaten daha önceden onaylandıysa (callback size ulaştıysa) sadece echo "OK"; exit; yaparak akışı sonlandırın.
    /* Ödeme durum sorgulama örnek
       $durum = SQL
       if($durum == "onay"){
            echo "OK";
            exit;
        }
     */

    if( $post['status'] == 'success' ) { ## Ödeme Onaylandı
        ## BURADA YAPILMASI GEREKENLER
        ## 1) Veri tabanınızda ödemeyi onaylayın.
        ## 2) Eğer müşterinize mesaj / SMS / e-posta gibi bilgilendirme yapacaksanız bu aşamada yapmalısınız.
        ## 3) $post['total_amount'] müşterinin yaptığı ödemenin toplam tutarıdır. Muhasebe işlemlerinizde
        ## bu tutraı kullanmanız gerekmektedir.
    } else {
        ## Link API'de başarısız ödemeler için callback yapılmamaktadır.
        ## Dolayısıyla kod akışında buraya erişim olmayacaktır. Ancak ileride Link API'de yapılabilecek geliştirmeler
        ## için dilerseniz buraya bir handler yazabilirsiniz.
    }

    ## Bildirimin alındığını PayTR sistemine bildir. OK yanıtını bu alandan kaldırmayın.
    echo "OK";
    exit;

File: docs/PayTR Linkle Ödeme/Türkçe/PayTR Link API - Callback (Optional)/Türkçe/Örnek PYTHON/PayTR Link API - Callback.py
# Python 3.6+
# Django Web Framework referans alınarak hazırlanmıştır
# ÖDEME LİNKİ BİLDİRİM ÖRNEK KODLAR

import base64
import hashlib
import hmac
import json

from django.shortcuts import render, HttpResponse
from django.views.decorators.csrf import csrf_exempt

@csrf_exempt
def callback(request):
    if request.method != 'POST':
        return HttpResponse(str(''))

    post = request.POST

    # API Entegrasyon Bilgileri - Mağaza paneline giriş yaparak BİLGİ sayfasından alabilirsiniz.
    merchant_key = b'XXXXXXXXXXXXXXXX'
    merchant_salt = 'XXXXXXXXXXXXXXXX'

    # Bu kısımda herhangi bir değişiklik yapmanıza gerek yoktur.
    # POST değerleri ile hash oluştur.
    hash_str = post['callback_id'] + post['merchant_oid'] + merchant_salt + post['status'] + post['total_amount']
    hash = base64.b64encode(hmac.new(merchant_key, hash_str.encode(), hashlib.sha256).digest())

    # Oluşturulan hash'i, paytr'dan gelen post içindeki hash ile karşılaştır
    # (isteğin paytr'dan geldiğine ve değişmediğine emin olmak için)
    # Bu işlemi yapmazsanız maddi zarara uğramanız olasıdır.
    if hash != post['hash']:
        return HttpResponse(str('PAYTR notification failed: bad hash'))

    """
    BURADA YAPILMASI GEREKENLER
    1) Ödeme durumunu post['callback_id'] değerini kullanarak veri tabanınızdan sorgulayın.
    2) Eğer ödeme zaten daha önceden onaylandıysa (callback size ulaştıysa) sadece 'OK' yaparak akışı sonlandırın.
    Ödeme durum sorgulama örnek
    durum = SQL
    
    if(durum == 'onay'){
         return HttpResponse(str('OK'))
    """

    if post['status'] == 'success':
        """
        BURADA YAPILMASI GEREKENLER
        1) Veri tabanınızda ödemeyi onaylayın.
        2) Eğer müşterinize mesaj / SMS / e-posta gibi bilgilendirme yapacaksanız bu aşamada yapmalısınız.
        3) post['total_amount'] müşterinin yaptığı ödemenin toplam tutarıdır. Muhasebe işlemlerinizde
        bu tutraı kullanmanız gerekmektedir.
        """
    else:
        """
        Link API'de başarısız ödemeler için callback yapılmamaktadır.
        Dolayısıyla kod akışında buraya erişim olmayacaktır. Ancak ileride Link API'de yapılabilecek geliştirmeler
        """

    # Bildirimin alındığını PayTR sistemine bildir.
    return HttpResponse(str('OK'))